### PR TITLE
DSD-438: Conditionally render "Required" or "Optional" in the `DatePicker`, `TextInput`, and `Label` components

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-438](https://jira.nypl.org/browse/DSD-438)

## This PR does the following:
- Uses `showOptReqLabel` prop to drill down whether the "Required" or "Optional" text should appear in a `Label` component.

## How has this been tested?

Locally and through Storybook.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr645-vxj9fp1o1pzklxsrwfrluh2qzrq5qtyv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/611d7deeb52cf94fcfbef0ac

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 
 - Exports `VideoPlayerAspectRatios`
 - Adds Tugboat QA configuration for PR preview deployments to replace Netlify.
+- Updates the `DatePicker`, `TextInput`, and `Label` components to pass an optional `showOptReqLabel` prop to conditionally render "Required"/"Optional" in the label text.
```
Comment (bigfishdesign13) [inline CHANGELOG.md:15]: This same pattern with `Label` is also used on `Select`, so `Select` will need to be listed here and the `Select` component will also need to be updated.

File: src/components/Label/Label.tsx [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
   const {
     attributes,
     blockName,
+    children,
     className,
     htmlFor,
     id,
-    modifiers = [],
     optReqFlag,
+    modifiers = [],
```
Comment (bigfishdesign13) [inline src/components/Label/Label.tsx:35]: Why was `modifiers` moved to the end of the list of props?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, this looks very good.  However, the same logic also needs to be added to the `Select` component.  The UI and labelling logic of `Select` mirrors that of `TextInput`.

File: src/components/Label/Label.tsx (hunk 2 above)
Comment (EdwinGuzman) [reply src/components/Label/Label.tsx:35]: Oh, not sure. I'll move it back.

File: CHANGELOG.md (hunk 1 above)
Comment (EdwinGuzman) [reply CHANGELOG.md:15]: That is true. I'd argue that it should be its own ticket to keep the scope of this PR for the `DatePicker` component. Updating the `Select` component to follow this pattern will not be difficult to do.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr645-vxj9fp1o1pzklxsrwfrluh2qzrq5qtyv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/611d7deeb52cf94fcfbef0ac

File: CHANGELOG.md (hunk 1 above)
Comment (bigfishdesign13) [reply CHANGELOG.md:15]: OK, makes sense.  I saw the new ticket for that task.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**